component/lambda/functions/upload-billing-hours.py

```diff
@@ -1,110 +1,106 @@
-from typing import Iterator, cast
+from typing import Iterator, NotRequired, cast
 from datetime import date, datetime, timedelta, timezone
 import itertools
-import boto3
+import logging
 
-from si_redshift import Redshift
-from si_lago_api import LagoApi, ExternalSubscriptionId, LagoEvent
-from si_logger import logger
-from si_types import OwnerPk, SqlTimestamp
+from si_lambda import SiLambda, SiLambdaEnv
+from si_lago_api import ExternalSubscriptionId, LagoEvent
+from si_types import SqlTimestamp
 
-
-session = boto3.Session()
-redshift = Redshift.from_env(session)
-lago = LagoApi.from_env(session)
-
-
-paul = OwnerPk("01GW0KXH4YJBWC7BTBAZ6ZR7EA")
 billable_metric_code = "resource-hours"
 cost_per_resource_hour = 0.007
```

Comment:
query: if we need to add a custom plan price for a user, are we able to do that / where are the user:pricing mappings (if anywhere)

Reply:
There are none--right now the system supports (a) the pay-as-you-go plan with its 100-concurrent-resources threshold; and (b) plans where the price is set in Lago and don't have that kind of threshold. Custom enterprise plans are not so likely to have a threshold where they go to zero, and this keeps us from having to maintain a big mapping for now.